plutus-core/testlib/PlutusCore/Generators/PIR/Common.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE BangPatterns          #-}
+{-# LANGUAGE ConstraintKinds       #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DeriveFunctor         #-}
+{-# LANGUAGE DerivingVia           #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE ImportQualifiedPost   #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NumericUnderscores    #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE PatternSynonyms       #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TupleSections         #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE UndecidableInstances  #-}
+{-# LANGUAGE ViewPatterns          #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+{-# OPTIONS_GHC -Wno-partial-type-signatures #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+
+module PlutusCore.Generators.PIR.Common where
+
+
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Data.String
+import Test.QuickCheck
+import Text.PrettyBy
+
+import PlutusCore.Default
+import PlutusCore.Name
+import PlutusIR
+
+-- CODE REVIEW: where to put the stuff below? Can we refactor to the point where we don't need them?
+-- Currently we need these for shrinking, getting rid of them would be nice.
+deriving stock instance Eq (Term TyName Name DefaultUni DefaultFun ())
+deriving stock instance Eq (Binding TyName Name DefaultUni DefaultFun ())
+deriving stock instance Eq (VarDecl TyName Name DefaultUni DefaultFun ())
+deriving stock instance Eq (TyVarDecl TyName ())
+deriving stock instance Eq (Datatype TyName Name DefaultUni DefaultFun ())
```

Comment:
Yeah, this seems quite suspicious. Is it even correct as it stands? Probably we want the alpha-equivalence versions.

Reply:
> Is it even correct as it stands?

I guess the shrinker isn't changing binders or something. Otherwise it may indeed rename a program to a different one that is "smaller" and alpha-equivalent to the original one.

Reply:
I've moved those instances to `GenerateTerms`, so that we don't need to worry about them right now. But we will need to worry about them later.